src/lab1/structures/plane/Plane.ts

```diff
@@ -1,19 +1,33 @@
+import { TraceableTransformable } from '../../../lab3/types/Transformable';
 import { Hit } from '../../types/Hit';
 import { Traceable } from '../../types/Traceable';
 import Normal3D from '../normal/Normal';
 import Ray from '../ray/Ray';
 import Vector3D from '../vector/Vector3D';
 import Vertex3D from '../vertex/Vertex3D';
 
-export default class Plane implements Traceable {
-  public readonly normal: Normal3D;
-  public readonly vertex: Vertex3D;
+export default class Plane implements TraceableTransformable {
+  public normal: Normal3D;
+  public vertex: Vertex3D;
 
   constructor(vertex: Vertex3D, normalVector: Vector3D) {
     this.normal = new Normal3D(normalVector);
     this.vertex = vertex;
   }
 
+  public rotate(angleX: number, angleY: number, angleZ: number): void {
+    this.normal = this.normal.rotate(angleX, angleY, angleZ);
+    this.vertex = this.vertex.rotate(angleX, angleY, angleZ);
+  }
+
+  public translate(x: number, y: number, z: number): void {
+    this.vertex = this.vertex.translate(x, y, z);
+  }
+
+  public scale(x: number, y: number, z: number): void {
+    this.vertex = this.vertex.scale(x, y, z);
```

Comment:
Scaling vertex? As far as I understood it is the save as translating this vertex. If so, this is unexpected behavior of scale method. Anyway, it's impossible to scale a plane, so this method shouldn't appear here